data/StateMachine.j2119

```diff
@@ -166,10 +166,18 @@ A Wait State MAY have a referencePath field named "TimestampPath".
 A Wait State MUST have only one of "Seconds", "SecondsPath", "Timestamp", and "TimestampPath".
 A Wait State MUST have a field named one of "Seconds", "SecondsPath", "Timestamp", or "TimestampPath".
 A Parallel State MUST have an object-array field named "Branches"; each element is a "Branch".
-A Map State MUST have an object field named "Iterator"; its value is a "Branch".
+A Map State MAY have an object field named "Iterator"; its value is a "Branch".
```

Comment:
`Iterator` should not be type `Branch`, it should also be type "Processor".

While you're here, maybe rename "Processor" to "ItemProcessor", for some extra precision in naming?